### PR TITLE
Rustls upgrade fix

### PR DESCRIPTION
**Pull request by nachog00**
This is the simplest solution that gets rid of the error:  #1389 

```bash
thread 'main' panicked at /home/chona/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustls-0.23.12/src/crypto/mod.rs:259:14:
no process-level CryptoProvider available -- call CryptoProvider::install_default() before this point
```

on `cargo run`.

I'm not sure if `ring` is the option we want nor if `run_cli` is the right place to call that install.

### CODE REVIEW

**Comment by zancas** [comment]
Have you run the full test suite against this feature locally?

If yes, did all the tests pass?

**Comment by zancas** [comment]
This is blocked on #1401

File: zingocli/src/lib.rs [hunk 1]
```diff
@@ -513,6 +514,11 @@ fn dispatch_command_or_start_interactive(cli_config: &ConfigTemplate) {
 
 /// TODO: Add Doc Comment Here!
```
Comment (zancas) [inline zingocli/src/lib.rs:515]: Upgrade the doc-comment, to something better.

Please add a doc-test.

File: zingocli/src/lib.rs [hunk 2]
```diff
@@ -513,6 +514,11 @@ fn dispatch_command_or_start_interactive(cli_config: &ConfigTemplate) {
...
 pub fn run_cli() {
+    // install default crypto provider (ring)
+    rustls::crypto::ring::default_provider()
+        .install_default()
+        .expect("Failed to install rustls crypto provider");
```
Comment (zancas) [inline zingocli/src/lib.rs]: In general `expect` and `unwrap` are not allowed in our codebase.  Should this function return Result?

**Review by zancas — DISMISSED** [review]

**Comment by fluidvanadium** [comment]
> This is blocked on #1401

not exactly. however
allowing Nacho to run the full test suite is maximum priority and will have to happen before we can expect him to perfect this fix

File: zingocli/src/lib.rs (hunk 1 above)
Comment (Oscar-Pepper) [reply zingocli/src/lib.rs:515]: this is the entrypoint to the CLI so i'm not sure we can add a doc-test for this fn

File: zingocli/src/lib.rs [hunk 3]
```diff
@@ -11,6 +11,7 @@ use std::sync::Arc;
 use log::{error, info};
 
 use clap::{self, Arg};
+use rustls;
```
Comment (Oscar-Pepper) [inline zingocli/src/lib.rs:14]: this line should not be necessary as you are calling `rustls::crypto::ring::default_provider().install_default()` with its full path

**Review by Oscar-Pepper — CHANGES_REQUESTED** [review]
Sorry I was not aware of the state of this PR when I wrote #1426. This looks good but we will also have to delete the crypto provider install from netutils as this is called multiple times which causes an error. Would you mind if we closed this for #1426 ?

**Comment by nachog00** [comment]
> Sorry I was not aware of the state of this PR when I wrote #1426. This looks good but we will also have to delete the crypto provider install from netutils as this is called multiple times which causes an error. Would you mind if we closed this for #1426 ?

I'm fine with closing this one. Didn't do it already yesterday only cause @zancas  asked me to add a test, documenting and some other stuff on #1426 and I wasnt sure how to collaborate on your PR. What is our flow in those cases? commit directly on it? PR on it? So far i got working on a local branch and was planning on PRing into yours.

**Comment by Oscar-Pepper** [comment]
> > Sorry I was not aware of the state of this PR when I wrote #1426. This looks good but we will also have to delete the crypto provider install from netutils as this is called multiple times which causes an error. Would you mind if we closed this for #1426 ?
> 
> I'm fine with closing this one. Didn't do it already yesterday only cause @zancas asked me to add a test, documenting and some other stuff on #1426 and I wasnt sure how to collaborate on your PR. What is our flow in those cases? commit directly on it? PR on it? So far i got working on a local branch and was planning on PRing into yours.

I commented on your PR as I think a doc test is difficult in this case but good practice for functions that can be unit tested. So in most cases the flow would be to PR into my branch. However, in this case I am happy for you to just PR into dev and I'll close mine or push directly to my branch if your have collaborator rights (I'm not sure if you being part of the organisation allows this by default). It's up to you really

**Comment by Oscar-Pepper** [comment]
closed for #1426

**Pull request closed without merging**